### PR TITLE
Fix Azure+HAProxy edge cases

### PR DESCRIPTION
**Pull request by jhunt**
Previously, activating the `haproxy` feature with an Azure CPI in play on your BOSH director would cause the availability set addons merge to occur, creating a stub `haproxy` instance_group (regardless of feature selection).  Later, when the `haproxy` feature was encountered, it would _insert_ a new `haproxy` instance group, after `nats`; this caused Spruce some severe issues later, because it could not handle two things named `haproxy`.

The hidden failure here is that _not_ specifying the `haproxy` feature would cause an haproxy instance group to appear anyway (because of the Azure availability set "override"), which would cause _BOSH_ to be quite unhappy.

Now, we have a baseline for Azure availability sets (that does not include haproxy) and another file for  overriding vm extensions if (and only if) haproxy is desired.

### CODE REVIEW

File: hooks/blueprint [hunk 1]
```diff
@@ -227,6 +223,16 @@ if want_feature small-footprint; then
   merge+=( manifests/addons/small-footprint.yml )
 fi
 
+case "$(bosh_cpi)" in
+(azure*)
+  merge+=( manifests/addons/azure.yml )
+  if want_feature haproxy; then
+    merge+=( maniests/addons/azure-with-haproxy.yml )
```
Comment (thomasmitchell) [inline hooks/blueprint]:
```suggestion
    merge+=( manifests/addons/azure-with-haproxy.yml )
```

File: hooks/blueprint [hunk 2]
```diff
@@ -227,6 +223,16 @@ if want_feature small-footprint; then
   merge+=( manifests/addons/small-footprint.yml )
 fi
 
+case "$(bosh_cpi)" in
+(azure*)
+  merge+=( manifests/addons/azure.yml )
+  if want_feature haproxy; then
+    merge+=( manifests/addons/azure-with-haproxy.yml )
```
Comment (thomasmitchell) [inline hooks/blueprint]:
```suggestion
    merge+=( manifests/azure-with-haproxy.yml )
```
or move the file to the addons folder.

Comment (jhunt) [reply hooks/blueprint]: Fixed?

**Comment by jhunt** [comment]
@krutten can you take this branch for a spin in a dev kit (without merging it or cutting a release quite yet)?

**Comment by dennisjbell** [comment]
Following up in PR# 50

**Pull request closed without merging**